packages/gestalt/src/DropdownItem.js

```diff
@@ -1,31 +1,63 @@
 // @flow strict
 import { type Node } from 'react';
-import OptionItem, { type OptionItemType } from './OptionItem.js';
+import OptionItem from './OptionItem.js';
 import { DropdownContextConsumer } from './DropdownContext.js';
 
-type PublicProps = {|
+type OptionItemType = {|
+  label: string,
+  subtext?: string,
+  value: string,
+|};
+
+type Props = {|
+  /**
+   * When supplied, will display a [Badge](https://gestalt.pinterest.systems/badge) next to the item's label. See the [Badges](https://gestalt.pinterest.systems/dropdown#Badges) variant to learn more.
+   */
   badgeText?: string,
+  /**
+   * If needed, users can supply custom content to each Dropdown Item. This can be useful when extra functionality is needed beyond a basic Link. See the [Custom item content](https://gestalt.pinterest.systems/dropdown#Custom-item-content) variant to learn more.
+   */
   children?: Node,
+  /**
+   * When supplied, will add a data-test-id prop to the dom element.
+   */
   dataTestId?: string,
-  onSelect: ({|
+  /**
+   * Callback when the user selects an item using the mouse or keyboard.
+   */ onSelect: ({|
     event: SyntheticInputEvent<HTMLInputElement>,
-    item: OptionItemType,
+    item: {|
+      label: string,
+      subtext?: string,
+      value: string,
+    |},
   |}) => void,
+  /**
+   * Object detailing the label, value, and optional subtext for this item.
+   */
   option: OptionItemType,
-  selected?: OptionItemType | $ReadOnlyArray<OptionItemType> | null,
-|};
-
-type PrivateProps = {|
-  index?: number,
-|};
-
-type Props = {|
-  ...PublicProps,
-  ...PrivateProps,
+  /**
+   * Either the selected item info or an array of selected items, used to determine when the "selected" icon appears on an item.
+   */
+  selected?:
+    | {|
+        label: string,
+        subtext?: string,
+        value: string,
+      |}
+    | $ReadOnlyArray<{|
+        label: string,
+        subtext?: string,
+        value: string,
+      |}>
+    | null,
+  /**
+   * Private prop used for accessibility purposes
+   */ index?: number,
```

Comment:
We previously did not document this prop. We should either filter it out manually, or (preferred) rename it to `_index` to indicate that it's a private prop, and so that it's filtered out automatically.

Reply:
Same, follow up PR